src/components/CreateDrinkForm.js

```diff
@@ -0,0 +1,464 @@
+import styled from 'styled-components';
+import { useForm } from 'react-hook-form';
+import { nanoid } from 'nanoid';
+import { useNavigate } from 'react-router-dom';
+import goBackArrow from '../icons/goBackArrow.svg';
+
+export default function CreateDrinkForm({ addNewDrink }) {
+  const navigate = useNavigate();
+
+  const {
+    register,
+    handleSubmit,
+    reset,
+    formState: { errors, isDirty },
+  } = useForm({
+    defaultValues: {
+      drinkName: '',
+      alcoholic: 'alcoholic',
+      category: 'Cocktail',
+      glass: 'Cocktail glass',
+    },
+  });
+
+  const onSubmit = data => {
+    const newDrink = {
+      idDrink: nanoid(),
+      isMyDrink: true,
+      strDrink: data.drinkName,
+      strCategory: data.category,
+      strAlcoholic: data.alcoholic,
+      strGlass: data.glass,
+      strInstructions: data.instructions,
+      strIngredient1: data.ingredient1,
+      strIngredient2: data.ingredient2,
+      strIngredient3: data.ingredient3,
+      strIngredient4: data.ingredient4,
+      strIngredient5: data.ingredient5,
+      strMeasure1: data.measure1,
+      strMeasure2: data.measure2,
+      strMeasure3: data.measure3,
+      strMeasure4: data.measure4,
+      strMeasure5: data.measure5,
+    };
+    console.log(newDrink);
+    addNewDrink(newDrink);
+    reset();
+    navigate(-1);
+  };
+
+  return (
+    <FormContainer>
+      <GoBackArrow onClick={() => navigate(-1)}>
+        <img src={goBackArrow} width="35" height="35" alt="go back arrow" />
+        <span className="sr-only">Go back arrow</span>
+      </GoBackArrow>
+      <StyledForm onSubmit={handleSubmit(onSubmit)}>
+        <FormHeadline>Create your own drink!</FormHeadline>
+        <label htmlFor="drinkName">
+          What is the name of your drink?
+          <FormInput
+            type="text"
+            id="drinkName"
+            placeholder="name"
+            autoFocus
+            {...register('drinkName', {
+              required: {
+                value: true,
+                message: 'Your drink needs a name',
+              },
+              minLength: {
+                value: 2,
+                message: 'This name is too short',
+              },
+              maxLength: {
+                value: 30,
+                message: 'This name is too long',
+              },
```

Comment:
You could put the `maxLength`as a prop to the input as well, otherwise the error message will appear but the use can still type in more than 30 characters
If you do that, remember to change the value of the maxLength for the error message to 29, otherwise the message will never appear

Reply:
Yes thanks, I will modify this like discussed